google/cloud/aiplatform/datasets.py

```diff
@@ -15,6 +15,358 @@
+                of the already existing one, and if labels with identical
+                key is imported before, the old label value will be
+                overwritten. If two DataItems are identical in the same
+                import data operation, the labels will be combined and if
+                key collision happens in this case, one of the values will
+                be picked randomly. Two DataItems are considered identical
+                if their content bytes are identical (e.g. image bytes or
+                pdf bytes). These labels will be overridden by Annotation
+                labels specified inside index file refenced by
+                [import_schema_uri][google.cloud.aiplatform.v1beta1.ImportDataConfig.import_schema_uri],
+                e.g. jsonl file.
+            import_schema_uri: Optional[str] = None
+                Points to a YAML file stored on Google Cloud
+                Storage describing the import format. Validation will be
+                done against the schema. The schema is defined as an
+                `OpenAPI 3.0.2 Schema
+                Object <https://tinyurl.com/y538mdwt>`__.
+            project: Optional[str]=None,
+                Project to upload this model to. Overrides project set in
+                aiplatform.init.
+            location: Optional[str]=None,
+                Location to upload this model to. Overrides location set in
+                aiplatform.init.
+            credentials: Optional[auth_credentials.Credentials]=None,
+                Custom credentials to use to upload this model. Overrides
+                credentials set in aiplatform.init.
+        Returns:
+            dataset (Dataset):
+                Instantiated representation of the managed dataset resource.
+        """
+
+        # Validate that source and import schema are passed together or not at all
+        if bool(source) ^ bool(import_schema_uri):
+            raise ValueError(
+                "Please provide both source and import_schema_uri to import data"
+            )
+
+        api_client = cls._instantiate_client(location=location, credentials=credentials)
+
+        create_dataset_lro = cls._create(
+            display_name=display_name,
+            parent=initializer.global_config.common_location_path(
+                project=project, location=location
+            ),
+            metadata_schema_uri=metadata_schema_uri,
+            request_metadata=metadata,
+            labels=labels,
+            api_client=api_client,
+        )
+
+        created_dataset = create_dataset_lro.result()
+
+        dataset_obj = cls(
+            dataset_name=created_dataset.name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        # If an import source was not provided, return empty created Dataset.
+        if not source:
+            return dataset_obj
+
+        return dataset_obj.import_data(
```

Comment:
This would return an LRO, while in the "no data import" case, we wait until the create_dataset LRO completes.  We need to document clearly (e.g. in the docstring) this inconsistent behavior.

Reply:
Yeah these multiple LROs under-the-hood can be misleading to the user. I'm adding this discussion to [b/171275584](http://b/171275584)